components/validator/validator.go

```diff
@@ -113,34 +119,46 @@ func NewValidator(parentCtx context.Context, cfg Config, l log.Logger, m metrics
 	}, nil
 }
 
-func (v *Validator) Start() {
+func (v *Validator) Start() error {
 	v.l.Info("starting Validator")
 
 	if !v.cfg.OutputSubmitterDisabled {
 		if err := v.l2os.Start(); err != nil {
-			v.l.Error("cannot start l2 output submitter: %w", err)
-			return
+			return fmt.Errorf("cannot start l2 output submitter: %w", err)
 		}
 	}
 
+	if err := v.challenger.Start(); err != nil {
+		return fmt.Errorf("cannot start challenger: %w", err)
+	}
```

Comment:
Don't we need to check the following?
```suggestion
	if !v.cfg.OutputSubmitterDisabled || !v.cfg.ChallengerDisabled {
		if err := v.challenger.Start(); err != nil {
			return fmt.Errorf("cannot start challenger: %w", err)
		}
	}
```
Or check it in config check function in `/components/validator/config.go` L46?
Also in `Stop()`.

Reply:
I got your point. 
I think validator should set `true` at least either `OutputSubmitterDisabled` or `ChallengerDisabled`. (That is, do check in config)
And the challenger should start whichever the mode is. wdyt?

Reply:
I do agree! Then it would be better to check if at least one of the flags is on in config.

Reply:
Added in [0cddd32](https://github.com/kroma-network/kroma/pull/59/commits/0cddd322255e32efd3865e0684a22515b1d8c1c5)